app/server/models/user.py

```diff
@@ -567,7 +567,7 @@ def has_valid_pin(self):
         self.clear_expired_pin_reset_tokens()
         not_resetting = len(self.pin_reset_tokens) == 0
 
-        return self.pin_hash is not None and not_resetting
+        return self.pin_hash is not None and not_resetting and self.failed_pin_attempts < 3
```

Comment:
now if a user is blocked, they can't use USSD at all until we reset their pin. otherwise it was ending up in some weird places like trying to reset pin even though they had 0 attempts left. didn't let them know they were blocked til the very end. 
would we want this? downside i can think of is if we have ussd menus that don't need pin, they can't access those anymore either

Reply:
This sounds reasonable to me. What do you think @enjeyw ?